php/php.editor/src/org/netbeans/modules/php/editor/model/impl/ClassScopeImpl.java

```diff
@@ -379,32 +372,145 @@ public Collection<? extends MethodScope> getInheritedMethods() {
     @Override
     public Collection<? extends FieldElement> getInheritedFields() {
         Set<FieldElement> allFields = new HashSet<>();
-        IndexScope indexScope = ModelUtils.getIndexScope(this);
-        ElementQuery.Index index = indexScope.getIndex();
-        org.netbeans.modules.php.editor.api.elements.ElementFilter filterForPrivate = org.netbeans.modules.php.editor.api.elements.ElementFilter.forPrivateModifiers(false);
+        Map<String, FieldElement> fieldNames = new HashMap<>();
         Set<ClassScope> superClasses = new HashSet<>(getSuperClasses());
         for (ClassScope classScope : superClasses) {
-            Set<org.netbeans.modules.php.editor.api.elements.FieldElement> indexedFields = filterForPrivate.filter(index.getAlllFields(classScope));
-            for (org.netbeans.modules.php.editor.api.elements.FieldElement field : indexedFields) {
-                allFields.add(new FieldElementImpl(classScope, field));
```

Comment:
This implementation has a bug:
All inherited fields are added to `classScope` as elements. 
If `classScope.getDeclaredFields()` is invoked, inherited fields may also be got.

https://github.com/apache/netbeans/blob/51fa1325cad00b4d9fc728ae207ec9657cabb31e/php/php.editor/src/org/netbeans/modules/php/editor/model/impl/ClassScopeImpl.java#L307-L318